pkg/kubernetes/scripts/virtual-machines/kube-middleware.jsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import { logDebug } from './utils.jsx';
+
+let kubeMethods = null;
+export function initMiddleware(_kubeMethods) {
+  kubeMethods = _kubeMethods;
+}
+
+async function vmShutdown({ vm }) {
+  const selfLink = vm.metadata.selfLink; // example: /apis/kubevirt.io/v1alpha1/namespaces/kube-system/virtualmachines/testvm
+  logDebug('vmShutdown(), selfLink: ', selfLink);
+  return await kubeMethods.delete(selfLink); // no value is reurned. An exception is thrown in case of failure
```

Comment:
what is the benefit of using async/await here?
`delete` already returns promise

Reply:
Good point. Two reasons:
- we want separate "business" logic from the frontend components
- the `shut down` is simple, we can expect more complex scenarios for next actions